src/adapter.ts

```diff
@@ -596,6 +599,43 @@ export class perlDebuggerConnection {
 		return result;
 	}
 
+	async getLoadedFiles(): Promise<string[]> {
+
+		const loadedFiles = await this.getExpressionValue(
+			`join "\t", grep { /^_</ } keys %main::`
+		);
+
+		return loadedFiles
+			.split(/\t/)
+			.filter(x => !/^_<\(eval \d+\)/.test(x))
+			.map(x => x.replace(/^_</, ''));
+
+	}
+
+	async getSourceCode(perlPath: string): Promise<string> {
+
+		// NOTE: `perlPath` must be a path known to Perl, there is
+		// no path translation at this point.
+
+		const escapedPath = perlPath.replace(
+			/([\\'])/g,
+			'\\$1'
+		);
```

Comment:
are we covered on windows?

Reply:
The `escapedPath` ends up as part of a string `eval`, the escaping is there to make sure "funny" characters will be interpreted as part of a string instead of arbitrary code.